src/lidar_subscriber.cpp

```diff
@@ -109,10 +103,19 @@ std::queue<Observation2D> laserscan_to_landmarks(const sensor_msgs::msg::LaserSc
     merge_landmarks(landmarks.front(), landmarks.back(), counts.front(), counts.back());
     landmarks.pop();
   }
-
-  #ifdef VISUALIZE_LANDMARK_OBSERVATIONS
-    std::queue<Observation2D>landmarks_copy(landmarks);
-    visualize_landmarks(landmarks_copy, observation_visualization_publisher);
-  #endif
-  return landmarks;
+  struct Observation2D curr_landmark;
```

Comment:
Moved wrapping to the end, once all of the points have been clustered into landmarks